### PR TITLE
fix: update ineff regex #357

### PR DESCRIPTION
**Pull request by Izook**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #357

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] Revision of an existing capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Comment by blackfalcon** [comment]
Using this brach, I built a test repo and this is how the demo page looks. 

Screen Shot 2023-07-06 at 4 46 42 PM

File: template/scripts/generateDocs.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ function formatTemplateFileContents(content, destination) {
   result = result.replace(/>(\r\n|\r|\n){2,}/g, '>\r\n'); // Remove empty lines directly after a closing html tag.
   result = result.replace(/>(\r\n|\r|\n)```/g, '>\r\n\r\n```'); // Ensure an empty line before code samples.
   result = result.replace(/>(\r\n|\r|\n){2,}```(\r\n|\r|\n)/g, '>\r\n```\r\n'); // Ensure no empty lines before close of code sample.
-  result = result.replace(/([^(\r\n|\r|\n)])(\r\n|\r|\n)+#/g, "$1\r\n\r\n#"); // Ensure empty line before header sections.
+  result = result.replace(/([^\r\n])([\r\n|\r|\n]+)/g, "$1\r\n\r\n#"); // Ensure empty line before header sections.
```
Comment (blackfalcon) [inline template/scripts/generateDocs.js:62]: This regex will malform the output of HTML from the markdown docs.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: template/scripts/generateDocs.js [hunk 2]
```diff
@@ -68,20 +79,15 @@ function formatTemplateFileContents(content, destination) {
...
-    .replace(/\r\n|\r|\n####\s`([a-zA-Z]*)`/g, `\r\n#### <a name="$1"></a>\`$1\`<a href="#${wcName}" style="float: right; font-size: 1rem; font-weight: 100;">back to top</a>`)
-    .replace(/\r\n|\r|\n\|\s`([a-zA-Z]*)`/g, '\r\n| [$1](#$1)')
-    .replace(/\| \[\]\(#\)/g, "");
-
   fs.writeFileSync(destination, result, { encoding: 'utf8'});
 
-  fs.readFile('./demo/apiExamples.md', 'utf8', function(err, data) {
-    formatTemplateFileContents(data, './demo/apiExamples.md');
+  fs.readFile('./demo/api.md', 'utf8', function(err, data) {
+    formatTemplateFileContents(data, './demo/api.md');
```
Comment (blackfalcon) [inline template/scripts/generateDocs.js:87]: I get the following error traced back to this update

```
./index.js → dist/...
created dist/ in 266ms
/Users/dalesande/src/auro/test-element/scripts/generateDocs.js:49
  result = result.replace(/\[npm]/g, nameExtractionData.npm);
                  ^

TypeError: Cannot read properties of undefined (reading 'replace')
    at formatTemplateFileContents (/Users/dalesande/src/auro/test-element/scripts/generateDocs.js:49:19)
    at ReadFileContext.callback (/Users/dalesande/src/auro/test-element/scripts/generateDocs.js:87:5)
    at FSReqCallback.readFileAfterOpen [as oncomplete] (node:fs:325:13)

Node.js v18.15.0
ERROR: "build:docs" exited with 1.
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: template/scripts/generateDocs.js (hunk 2 above)
Comment (Izook) [reply template/scripts/generateDocs.js:87]: In what repo/branch did you see this?

Comment (blackfalcon) [reply template/scripts/generateDocs.js:87]: I generated a new repo using `npm run build:test`

**Pull request closed without merging**